### PR TITLE
Added request sharing functionality.

### PR DESCRIPTION
**Pull request by Penguin-12**

### CODE REVIEW

File: lib/home/model/Request.dart [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    double userLongitude = userLocation.longitude;
+    double latDifference = pi * (userLatitude - requestLatitude).abs() / 180;
+    double lonDifference = pi * (userLongitude - requestLongitude).abs() / 180;
+    double a = sin(latDifference / 2) * sin(latDifference / 2) +
+        cos(pi * (requestLatitude) / 180) *
+            cos(pi * (userLatitude) / 180) *
+            sin(lonDifference / 2) *
+            sin(lonDifference / 2);
+    double c = 2 * atan2(sqrt(a), sqrt(1 - a));
+    double distance = 6378 * c * 1000;
```
Comment (PrarabdhGarg) [inline lib/home/model/Request.dart]: Could you check once, there must be some library function or package to calculate the distance

File: lib/home/model/RequestsController.dart [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import 'package:fluttertoast/fluttertoast.dart';
+import 'package:location/location.dart';
+import 'package:nearby_connections/nearby_connections.dart';
+
+class RequestsController extends ChangeNotifier {
+  final FlutterSecureStorage _secureStorage;
+  final Nearby _nearby = Nearby();
+  final Location _location = Location();
+  final Strategy _strategy = Strategy.P2P_CLUSTER;
+  final List<Request> requests = [];
```
Comment (PrarabdhGarg) [inline lib/home/model/RequestsController.dart:13]: Error handling is missing. But we could probably leave this for now

File: pubspec.yaml [hunk 3]
```diff
@@ -32,7 +32,8 @@ dependencies:
   flutter_secure_storage: ^3.3.3
   fluttertoast: ^4.0.0
   location: ^3.0.0
-
+  nearby_connections: ^1.1.1+1
+
```
Comment (PrarabdhGarg) [inline pubspec.yaml:37]: Please remove this extra line

**Review by PrarabdhGarg — CHANGES_REQUESTED** [review]

File: lib/home/model/RequestsController.dart (hunk 2 above)
Comment (Penguin-12) [reply lib/home/model/RequestsController.dart:13]: Ya i was planning to do handle errors when the whole request functionality is complete or somewhat stable as i was still discovering how to implement the library and hence there were many changes.

**Pull request merged**